core/services/headtracker/head_tracker.go

```diff
@@ -131,17 +134,19 @@ func (ht *HeadTracker) Start() error {
 	})
 }
 
-func (ht *HeadTracker) getInitialHead() (models.Head, error) {
+func (ht *HeadTracker) getInitialHead() (*models.Head, error) {
 	ctx, cancel := eth.DefaultQueryCtx()
 	defer cancel()
 	head, err := ht.ethClient.HeadByNumber(ctx, nil)
 	if err != nil {
-		return models.Head{}, errors.Wrap(err, "failed to fetch initial head")
-	} else if head == nil {
-		return models.Head{}, errors.New("initial head was unexpectedly nil")
+		return nil, errors.Wrap(err, "failed to fetch initial head")
 	}
-	ht.logger().Debugw("HeadTracker: got initial current block", "blockNumber", head.Number, "blockHash", head.Hash)
-	return *head, nil
+	loggerFields := []interface{}{"head", head}
```

Comment:
hm this will print something likely not very readable? maybe better to print a separate message for nil head?

Reply:
I was going for the approach of "the more info the better". We also log blockNumber and blockHash, since its structured anyway it won't clutter things